### ISSUE TITLE
Add Gradient Clipping

### ISSUE THREAD

**Issue by vitalwarley** [vitalwarley/research#10]
[How-to](https://pytorch-lightning.readthedocs.io/en/latest/advanced/training_tricks.html#gradient-clipping
). [Motivation](https://github.com/deepinsight/insightface/blob/17cdeab12a35efcebc2660453a8cbeae96e20950/recognition/arcface_torch/train.py#L139).

**Comment by vitalwarley** [comment]
Basically,

```
Trainer(gradient_clip_val=5)
```

`norm` is the default algorithm, as originally: `torch.nn.utils.clip_grad_norm_(backbone.parameters(), 5)`

**Comment by vitalwarley** [comment]
`--gradient_clip_val 5` na CLI.

**Comment by vitalwarley** [comment]
For clarification, I didn't change `gradient_clip_algorithm` to `value` (see why I would [here](https://pytorch-lightning.readthedocs.io/en/1.5.4/advanced/training_tricks.html)) because [insightface](https://github.com/deepinsight/insightface/blob/e78eee59caff3515a4db467976cf6293aba55035/recognition/arcface_torch/train.py#L160) uses `clip_grad_norm`.
